src/devices/src/virtio/block/event_handler.rs

```diff
@@ -27,25 +58,111 @@ impl Subscriber for Block {
         let supported_events = EventSet::IN;
         if !supported_events.contains(event_set) {
             warn!(
-                "Received unknown event: {:?} from source: {:?}",
+                "Block: Received unknown event: {:?} from source: {:?}",
                 event_set, source
             );
             return;
         }
 
-        // Looks better than C style if/else if/else.
-        match source {
-            _ if queue_evt == source => self.process_queue_event(),
-            _ if rate_limiter_evt == source => self.process_rate_limiter_event(),
-            _ => warn!("Spurious event received: {:?}", source),
-        }
+        match self.device_state {
+            DeviceState::Activated(_) => {
+                let queue_evt = self.queue_evts[0].as_raw_fd();
+                let rate_limiter_evt = self.rate_limiter.as_raw_fd();
+                let activate_fd = self.activate_evt.as_raw_fd();
+
+                // Looks better than C style if/else if/else.
+                match source {
+                    _ if queue_evt == source => self.process_queue_event(),
+                    _ if rate_limiter_evt == source => self.process_rate_limiter_event(),
+                    _ if activate_fd == source => self.process_activate_event(evmgr),
+                    _ => warn!("Block: Spurious event received: {:?}", source),
+                }
+            }
+            DeviceState::Inactive => warn!(
+                "Block: The device is not yet activated. Spurious event received: {:?}",
+                source
+            ),
+        };
     }
 
     // Returns the rate_limiter and queue event fds.
     fn interest_list(&self) -> Vec<EpollEvent> {
-        vec![
-            EpollEvent::new(EventSet::IN, self.rate_limiter.as_raw_fd() as u64),
-            EpollEvent::new(EventSet::IN, self.queue_evts[0].as_raw_fd() as u64),
-        ]
+        vec![EpollEvent::new(
+            EventSet::IN,
+            self.activate_evt.as_raw_fd() as u64,
+        )]
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use std::sync::{Arc, Mutex};
+
+    use super::*;
+    use crate::virtio::block::device::tests::*;
+    use crate::virtio::device::VirtioDevice;
+    use crate::virtio::queue::tests::*;
+    use virtio_gen::virtio_blk::*;
+    use vm_memory::{Bytes, GuestAddress};
+
+    #[test]
+    fn test_interest_list() {
```

Comment:
This test does not bring much value because `interest_list` just returns a static vector created in the function body. To test this basically means to duplicate the code in the test. This is typically a bad engineering practice referred to as WET. You can read more about it here: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
I agree we should be consistent. If people from the team agree with this, we shouldn't introduce this test in this PR just for the sake of consistency though :D

Reply:
I've removed the WET tests.